### DISCUSSION TITLE
NBT Data with Identities

### DISCUSSION THREAD

**Discussion by PsychoBreadstick** [Draylar/identity#358] [Ideas]
As most of us know, you can assign nbt data to entities using the /summon command. Allowing for you to summon entities with specific data. Such as spawning a horse with specific colors. Or spawning a bee that already has nectar on it.

I was thinking about the idea of being able to do something similar with the /identity equip command. Allowing you to give yourself or another player an identity with specific data. Here are a few examples:

"/identity equip @s minecraft:bee {HasNectar:1,Glowing:1b}" Which would equip a bee identity on you that has nectar on it, and also has a glowing outline.

"/identity equip @p minecraft:horse {Variant:771,SaddleItem:{id:saddle,Count:1},Invulnerable:1b}" Which would equip a horse identity on you that is brown with white dote on it, equipped with a saddle, while also being invulnerable.

"/identity equip @p minecraft:llama {Variant:1,DecorItem:{id:red_carpet,Count:1},Invulnerable:1b}" Which would equip a llama identity on you that is white and is equipped with a red carpet.

Of course, not all nbt data will work, such as NoAI due to the fact that identities can't have AI since they are controlled by players, PersistenceRequired due to the fact that you can't despawn, and a lot of other nbt data.

Although, the main idea of this is to allow for more customizability with identity for mods/datapacks using it. By allowing identities to contain unique nbt data just like how the /summon command does already by allowing to spawn mobs with nbt data.

**Comment by ItsSnowia** [comment]
I feel this could replace the need to add more mob variants prior to identity 2.3.0, as you can for example as an wolf disguise have a Yellow collar with NBT Tag for the wolf "{CollarColor:11}" and could be very useful for more advanced players with NBT Tag knowledge, but although not fully user accessible by everyone since not everyone doesn't know how NBT Tags work, but for me I just been waiting for something like identity to be able to turn into a mob but as-well provide the ability to actually apply NBT tags to your disguise.
